libcxx/include/string

```diff
@@ -2144,6 +2119,58 @@ private:
     return __is_long() ? __get_long_pointer() : __get_short_pointer();
   }
 
```

Comment:
Perhaps adding commented sections like

```
//////////////////////////////////////////////////////////////
// Internal buffer management
//////////////////////////////////////////////////////////////

//////////////////////////////////////////////////////////////
// Internal representation
//////////////////////////////////////////////////////////////
```

would help organize things a bit here without requiring full blown encapsulation in a class.